### PR TITLE
access w as an item DiscreteField not an attribute ndarray

### PR DESCRIPTION
**Pull request by gdmcbain**
For #548.

### CODE REVIEW

**Comment by gdmcbain** [comment]
I see that the item-style access is consistent with

https://github.com/kinnala/scikit-fem/blob/2957f0443be445d975c0299b9bef02ed4e2f5886/docs/examples/ex10.py#L14-L23

https://github.com/kinnala/scikit-fem/blob/2957f0443be445d975c0299b9bef02ed4e2f5886/docs/examples/ex13.py#L54-L57

https://github.com/kinnala/scikit-fem/blob/2957f0443be445d975c0299b9bef02ed4e2f5886/docs/examples/ex22.py#L47-L54

File: skfem/models/general.py [hunk 1]
```diff
@@ -11,9 +11,9 @@ def divergence(u, v, w):
 
 @LinearForm
 def rot(v, w):
-    return dot(curl(v), w.w)
+    return dot(curl(v), w["w"])
```
Comment (kinnala) [inline skfem/models/general.py:14]: Now that we have the option to use pretty much any name in place of `"w"` (I belive when this was originally implemented, it was only `"w"` allowed), the choice of `"w"` feels rather arbitrary. I wonder if we could improve these forms somehow by keeping that in mind?

Comment (kinnala) [reply skfem/models/general.py:14]: See #550

Comment (gdmcbain) [reply skfem/models/general.py:14]: I see, yes, that's a good idea, a less arbitrary name than `"w"`.  But then how is the user to know or remember it?  Perhaps better still is indeed the [tacit](https://en.wikipedia.org/wiki/Tacit_programming) style enabled by #550. Does that mean deprecating `rot` and `vrot`?  I suppose that if they're left in `skfem.models.general` at least temporarily during deprecation then the attribute &rarr; item fix is still required.

I think `rot` needs a different `BilinearForm` to `vrot`.

**Comment by gdmcbain** [comment]
This is already covered by e19e5810771b00c9145b69b21715ac0fe267372e, though the [ideas](https://github.com/kinnala/scikit-fem/pull/549#discussion_r569272259) of either coming up with a better name for `"w"` or avoiding having to name it are worth considering.

**Pull request closed without merging**